### PR TITLE
Add EIP: Interface Naming Conventions

### PR DESCRIPTION
**Pull request by pcaversaccio**
This specification defines a standardised way of naming interface functions, event or custom error definitions, or anything else included in an interface definition that is used as an identifier by compilers in some form.

### CODE REVIEW

File: EIPS/eip-canonicalisation_naming_conventions.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+---
+eip:
```
Comment (abcoathup) [inline EIPS/eip-canonicalisation_naming_conventions.md:2]:
```suggestion
eip: 7182
```
Assigning PR as EIP number, please also update the filename

**Comment by pcaversaccio** [comment]
quickly following up: any feedback @axic, @Pandapip1, @SamWilsn, @xinbenlv or can we get this merged since all tests pass?

**Comment by SamWilsn** [comment]
I am closing this pull request because we are in the process of separating EIPs and ERCs into distinct repositories. Unfortunately, as far as we are aware, GitHub does not provide any tools to ease this migration, so every pull request will need to be re-opened manually.

As this is a PR to create / modify an ERC, I will kindly ask you to redirect this to the new repository at [ethereum/ERCs](https://github.com/ethereum/ERCs). **We have prepared [a guide](https://github.com/ethereum/ERCs/issues/8) to help with the process.**

If there is relevant history here, please link to this PR from the new pull request.

On behalf of the EIP Editors, I apologize for this inconvenience.

**Comment by SamWilsn** [comment]
@pcaversaccio Sorry it's taken so long to get someone to take a look at this! From a first pass, I would say that this isn't a good fit for an EIP. I'd probably publish this as a standalone style guide.

---

Unless I'm misunderstanding the point? Maybe this is supposed to be a normalization step that compilers perform before generating the ABI and hashing interface ids / function selectors? If this is something that is supposed to be shared by all compilers, (1) I think that needs to be made more clear in the proposal, and (2) wouldn't this introduce a ton of backwards compatibility issues?

**Pull request closed without merging**